priv/perf/apps/load_test/lib/service/faucet.ex

```diff
@@ -45,48 +69,124 @@ defmodule LoadTest.Service.Faucet do
+    state =
+      struct!(
+        __MODULE__,
+        faucet_account: faucet_account,
+        fee: Keyword.fetch!(config, :fee_wei)
+      )
 
-    state = struct!(__MODULE__, account: faucet, fee: fee_wei, utxos: %{@eth => eth_utxo})
     {:ok, state}
   end
 
-  def handle_call(:get_faucet, _from, %__MODULE__{account: faucet} = state) do
-    {:reply, {:ok, faucet}, state}
+  def handle_call(:get_faucet, _from, %__MODULE__{faucet_account: faucet_account} = state) do
+    {:reply, {:ok, faucet_account}, state}
   end
 
-  def handle_call({:fund_child_chain, account, amount, @eth = token}, _from, %__MODULE__{account: faucet} = state) do
-    {utxo, utxo_amount} = state.utxos[token]
-    change = utxo_amount - amount - state.fee
-    if change < 0, do: raise({:error, :change_below_zero})
+  def handle_call(
+        {:fund_child_chain, receiver, amount, currency},
+        _from,
+        %__MODULE__{
+          faucet_account: faucet_account
+        } = state
+      ) do
+    utxo = check_sufficient_funds(state, currency, amount)
 
-    Logger.debug("Funding user: #{Encoding.to_hex(account.addr)} with UTXO: #{Utxo.pos(utxo)}")
+    change = utxo.amount - amount - state.fee
+    if change < 0, do: raise({:error, :insufficient_faucet_funds})
 
     outputs = [
-      %Utxo{amount: change, currency: token, owner: faucet.addr},
-      %Utxo{amount: amount, currency: token, owner: account.addr}
+      %Utxo{amount: change, currency: currency, owner: faucet_account.addr},
+      %Utxo{amount: amount, currency: currency, owner: receiver.addr}
     ]
 
-    {:ok, blknum, txindex} = Transaction.submit_tx([utxo], outputs, [faucet], @fund_child_chain_account_retries)
-    {:ok, change_utxo} = Utxo.new(%{blknum: blknum, txindex: txindex, oindex: 0})
-    {:ok, user_utxo} = Utxo.new(%{blknum: blknum, txindex: txindex, oindex: 1, amount: amount})
+    Logger.debug("Funding user #{Encoding.to_hex(receiver.addr)} with #{amount} from utxo: #{Utxo.pos(utxo)}")
+    {:ok, blknum, txindex} = Transaction.submit_tx([utxo], outputs, [faucet_account], @fund_child_chain_account_retries)
 
-    updated_state = Map.put(state, :utxos, %{state.utxos | token => {change_utxo, change}})
+    next_faucet_utxo = %Utxo{blknum: blknum, txindex: txindex, oindex: 0, amount: change}
+    user_utxo = %Utxo{blknum: blknum, txindex: txindex, oindex: 1, amount: amount}
+
+    updated_state = Map.put(state, :utxos, Map.put(state.utxos, currency, next_faucet_utxo))
 
     {:reply, {:ok, user_utxo}, updated_state}
   end
 
-  defp get_funded_faucet_account(opts) do
-    faucet_private_key = Keyword.fetch!(opts, :faucet_private_key)
-    {:ok, faucet} = Account.new(faucet_private_key)
+  @spec check_sufficient_funds(state(), Utxo.address_binary(), pos_integer()) :: Utxo.t()
+  defp check_sufficient_funds(
+         %{utxos: utxos, faucet_account: faucet_account, fee: fee},
+         currency,
+         amount
+       ) do
+    utxo =
+      case Map.has_key?(utxos, currency) do
```

Comment:
I would prefer `if` instead of `case`. Feels more natural when dealing with boolean values.

Reply:
not according to https://erlef.github.io/security-wg/secure_coding_and_deployment_hardening/elixir_truthy
```
Prefer case over if, unless or cond
```

Reply:
I have this question for a long while. But why in the case of boolean `cond` would be better option (I don't see explanation on this topic by the link you provide)? Also when should you use if?

Reply:
That's the worst thing ever. I've been doing ifs since grade school.

Reply:
`Map.has_key` returns a boolean `has_key?(map(), key()) :: boolean()` so I think that's not  `truthy`. Do we follow a convention to always use `case` no matter what?

Reply:
I haven't really found a good explanation _why_ to use case over if either.

I can understand the reasoning behind reducing conditionals in functional programming (e.g. http://degoes.net/articles/destroy-all-ifs)

But sometimes I believe it's not possible to avoid them. In those cases, I find the `if` construct much more readable than `case`

Having said that, I also find 
```
fn x -> x["amount"] end
```
more readable than
```
& &1["amount"]
```
so I guess a certain amount of it is just what you're used to.

Reply:
Commit https://github.com/omisego/elixir-omg/pull/1445/commits/c45f1468f18ab3ce57aaa687921a29fc28defa02
gets rid of an `if` in favour of a more functional approach.

Is that way better?

Reply:
interesting points all of you. which really points to our loose loose `code style` definition. 
I prefer `cases` almost 90% of the time, for me, everything IS pattern matching (well... it is, even if you use an `if`). for what is worth, there are no variable assignments either, that too is pattern matching.

I want to avoid us having these discussions/these sort of PR reviews in the future. we can do the following:
1. we don't comment on peoples usage of conditional statements
2. we agree on a principal, write it into our code style document and we follow it as closely as possible. from this point onward, we try not to comment of conditional statement.

Second point.
elixir-omg is greatly suffering from readability because of excessive usage of `capture syntax`. it's nice and all, but SHORT/expressive code does not mean BETTER code almost always. Why?
Because 90% of the time you're not writing code, but you're reading code. Not only you, others.